pydevices/acq400Devices/acq400_base.py

```diff
@@ -142,7 +164,21 @@ def __init__(self,dev):
                 self.chans.append(getattr(self.dev, 'INPUT_%3.3d'%(i+1)))
                 self.decim.append(getattr(self.dev, 'INPUT_%3.3d:DECIMATE' %(i+1)).data())
             self.seg_length = self.dev.seg_length.data()
-            self.segment_bytes = self.seg_length*self.nchans*np.int16(0).nbytes
+            self.segment_bytes = self.seg_length*self.nchans*numpy.int16(0).nbytes
+
+            #Fechting all calibration information from every channel. Save it in INPUT_XXX:CAL_INPUT
+            uut.fetch_all_calibration()
+            eslo = uut.cal_eslo[1:]
+            eoff = uut.cal_eoff[1:]
+            channel_data = uut.read_channels()
+
+            for ic, ch in enumerate(self.chans):
+                if ch.on:
+                    ch.EOFF.putData(float(eoff[ic]))
+                    ch.ESLO.putData(float(eslo[ic]))
+                    expr = "{} * {} + {}".format(ch, ch.ESLO, ch.EOFF)
```

Comment:
well all of em

Reply:
Add you forgot to put the actual data in ch.. beast remove the EOFF ESLO and CAL_stuff and instead make teh data a signal with 
```
for ic, ch in enumerate(self.chans):
    if ch.on:
        cal = MDSplus.TdiCompile("$VALUE * $ + $", float(eslo[ic]), float(eoff[ic]))
        ch.record = MDSplus.Signal(cal, chan_data[ci])
```